app/shop.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+import dataclasses
+from typing import List, Optional
+
+from app.customer import Customer
+
+
+@dataclasses.dataclass
+class Shop:
+    name: str
+    location: List[int]
+    products: dict
+
+    def total_price(self, product_cart: dict) -> float:
+        price = 0
+        for product, count in product_cart.items():
+            price += self.products[product] * count
+
+        return round(price, 2)
+
+    def list_read(self: List[dict]) -> list:
+
+        return [Shop(**shop) for shop in self]
```

Comment:
Iterate by self is looks strange for me. I would prefer more obvious code like store all instance it list or dict and iterate by it